src/plugins/orangehrmOpenidAuthenticationPlugin/Service/SocialMediaAuthenticationService.php

```diff
@@ -19,16 +19,21 @@
 
 namespace OrangeHRM\OpenidAuthentication\Service;
 
-use Jumbojett\OpenIDConnectClient;
 use Jumbojett\OpenIDConnectClientException;
 use OrangeHRM\Authentication\Dto\UserCredential;
 use OrangeHRM\Entity\AuthProviderExtraDetails;
 use OrangeHRM\OpenidAuthentication\Dao\AuthProviderDao;
+use OrangeHRM\OpenidAuthentication\OpenID\OpenIDConnectClient;
+use OrangeHRM\OpenidAuthentication\Traits\Service\SocialMediaAuthenticationServiceTrait;
 
 class SocialMediaAuthenticationService
 {
+    use SocialMediaAuthenticationServiceTrait;
     private AuthProviderDao $authProviderDao;
 
+    public const SCOPE = 'email';
+    public const REDIRECT_URL = 'https://734d-2402-d000-a500-40f9-f1e8-1109-5f81-bcf4.ngrok-free.app/openidauth/openIdCredentials';
```

Comment:
TODO:: This redirect url generation logic should be there in a single place

Reply:
added to service